src/garage/_dtypes.py

```diff
@@ -227,9 +219,20 @@ def __new__(cls, env_spec, observations, last_observations, actions,
                     'length {}, but got key {} with batch size {} instead.'.
                     format(inferred_batch_size, key, val.shape[0]))
 
+        # step_types
```

Comment:
can you check that `lengths` is consistent with `step_types`?

Reply:
If I understand your question correctly - `inferred_batch_size` is from `lengths` and this is checking if the length of `step_types` is consistent with `lengths`.